trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,228 @@
+    if classification and noise_free:
+        gpflow.set_trainable(model.kernel.variance, False)
+
+    return model
+
+
+def build_svgp(
+    data: Dataset,
+    search_space: Box,
+    classification: bool = False,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+    num_inducing_points: Optional[int] = None,
+    trainable_inducing_points: bool = False,
+) -> SVGP:
+    empirical_mean, empirical_variance, num_data_points = _get_data_stats(data)
+
+    if classification:
+        if noise_free:
+            empirical_variance = 100.0
+        else:
+            empirical_variance = 1.0
+        model_likelihood = gpflow.likelihoods.Bernoulli()
+    else:
+        if likelihood_variance is None:
+            noise_variance = 0.1 * empirical_variance
+        else:
+            noise_variance = likelihood_variance
+        model_likelihood = gpflow.likelihoods.Gaussian(noise_variance)
+
+    kernel = _get_kernel(
+        empirical_variance, search_space, kernel_priors, classification, noise_free
+    )
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    model = SVGP(
+        kernel,
+        model_likelihood,
+        inducing_points,
+        mean_function=mean,
+        num_data=num_data_points,
+    )
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+    if classification:
+        gpflow.set_trainable(model.kernel.variance, False)
+
+    return model
+
+
+def _get_data_stats(data: Dataset) -> tuple[TensorType, TensorType, int]:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    empirical_mean = tf.math.reduce_mean(data.observations)
+    num_data_points = len(data.observations)
+
+    return empirical_mean, empirical_variance, num_data_points
+
+
+def _get_kernel(
+    variance: TensorType,
+    search_space: Box,
+    priors: bool,
+    classification: bool = False,
+    noise_free: bool = False,
+) -> gpflow.kernels.Kernel:
+    dim = search_space.upper.shape[-1]
+    ub, lb = search_space.upper[-1], search_space.lower[-1]
+
+    lengthscales = [0.2 * (ub - lb) * math.sqrt(dim)] * dim
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=lengthscales)
+
+    if priors:
+        prior_scale = tf.cast(1.0, dtype=gpflow.default_float())
+        if not (classification and noise_free):
+            kernel.variance.prior = tfp.distributions.LogNormal(tf.math.log(variance), prior_scale)
+        kernel.lengthscales.prior = tfp.distributions.LogNormal(
+            tf.math.log(kernel.lengthscales), prior_scale
+        )
+
+    return kernel
+
+
+def _get_inducing_points(search_space: Box, num_inducing_points: Optional[int]) -> TensorType:
+    if num_inducing_points is None:
+        num_inducing_points = min(500, 25 * search_space.upper.shape[-1])
```

Comment:
What is the "25 * search_space.upper.shape[-1]" term motivated by?

Reply:
@vpicheny?

Reply:
Motivated by... the fact that 50 /75 points seem reasonable in 2D / 3D. And we want this number to grow with dimension, but not took quickly.

But I guess any other heuristic number would do!!!

Reply:
I see. I'd be tempted to go min(500, n_data), or do we not know n_data at this stage? If we wanted something smaller for speed at low dims, I'd suggest being a bit more aggressive on the dimension scaling like D**2.

Reply:
This is at initialisation stage so we wouldn't want to use `n_data`, because BO might start with very few points. And we don't want to change this number later on, to avoid retracing. 

D**2 gives us 400 points for D=4, feels to me like a bit much??

Reply:
I've never tried using sobol samples as inducing points, but doesn't it have the risk that if they are not trainable your BO loop will become stuck (it can't reduce variance where its selecting the latest data)? It seems suboptimal on small datasets on which BO is usually run, in which case you may have enough inducing points to put them on the data exactly?

Reply:
I have opted for keeping the original numbers, seems ok for now